src/tests/ie_test_utils/common_test_utils/test_assertions.hpp

```diff
@@ -5,6 +5,7 @@
 #pragma once
 
 #include <gtest/gtest.h>
+#include <gmock/gmock-matchers.h>
```

Comment:
It can be reverted

Reply:
`EXPECT_THAT` (https://github.com/openvinotoolkit/openvino/blob/974650ba6d7a0a17cddcfd944c285e47663f919b/src/tests/ie_test_utils/common_test_utils/test_assertions.hpp#L82) macros is from <gmock/gmock-matchers.h>. So each target which uses `OV_EXPECT_THROW` (based on `EXPECT_THAT`) should be linked with `gmock` by itself.

In this case I decided to add this dependency directly to  `commonTestUtils` target

Reply:
if want (and is really required here) include 'gmock' use  `#include <gmock/gmock.h>`  instead, and include `gtest` shouldn't be required as `gmock` depends on it same for linking

Reply:
Include full gmock.h for easier extension later

Reply:
Reverted - gmock.h also introduces `Assign` class which conflicts with our `Assign` op, so it requires additional changes.
Anyway, I would prefer to leave as is in order to not include huger header because purpose is obscure